spec/GraphQLOverHTTP.md

```diff
@@ -201,8 +201,17 @@ may not be the GraphQL server but instead some intermediary such as API gateways
 
 ## Status Codes
 
-If the `data` entry in the response has any value other than `null` (when the operation has successfully executed
-without error) then the response should use the 200 (OK) status code.
-If the operation failed before or during execution, due to a syntax error, missing information, validation error
-or any other reason, then response should use the 4XX or 5XX status codes.
-It is recommended to use the same error codes as the [reference implementation](https://github.com/graphql/express-graphql).
+If the response has Content-Type GraphQL and contains a non-null `data` entry,
+then it MUST have status code `2xx`, and it SHOULD have status code `200`.
+
+If the response has Content-Type GraphQL and has a non-`2xx` status code,
+the `data` entry must be either:
+- equal to `null` (in case of an execution error)
+- not present (in case of a validation error)
```

Comment:
The value of `data` is fully covered by GraphQL specification I don't think we should duplicate it here.
Instead, just specify it should be `null` or not present at all.